apiserver/common/applicationwatcher.go

```diff
@@ -157,6 +155,13 @@ func (w *applicationWatcher) handle(changes []string) ([]string, error) {
 	return filteredChanges, nil
 }
 
+func metadataFormatV2(manifest *charm.Manifest) charm.Format {
+	if manifest != nil && len(manifest.Bases) > 0 {
```

Comment:
@SimonRichardson and I were discussing how the V1 and V2 shouldn't be leaking out of the charm package.

Reply:
let's do a follow up